Runtime/Interaction/CustomMeshUtils/CustomMeshCollidersManager.cs

```diff
@@ -0,0 +1,144 @@
+using System;
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Jobs;
+using UnityEngine;
+using UnityEngine.Jobs;
+
+namespace ubco.ovilab.HPUI.CustomMeshUtils
+{
+    /// <summary>
+    /// Component to manage the colliders for a given rectangular custom mesh, based on <see cref="Interaction.DeformableSurfaceCollidersManager"/>
+    /// </summary>
+    public class CustomMeshCollidersManager : MonoBehaviour
+    {
+        [SerializeField] private VertexRemapData vertexRemapData;
+
+        // [SerializeField] private GameObject rectifiedVertexDS;
+        // [SerializeField, Range(0, 120)] private int id;
+        
+        private NativeArray<Vector3> vertices_native, normals_native;
+        private List<Vector3> vertices = new List<Vector3>(), normals = new List<Vector3>();
+        private NativeArray<int> remapped_vertices_data;
+        private TransformAccessArray colliderObjects;
+        private Mesh tempMesh;
+        private bool generatedColliders;
+        private SkinnedMeshRenderer targetMesh;
+        private float scaleFactor = 0.001f;
+        private int meshXRes, meshYRes;
+        
+        private void Update()
+        {
+            if (!generatedColliders || !isActiveAndEnabled) return;
+
+            UpdateColliderPositions();
+            // rectifiedVertexDS.transform.localPosition = vertices[remappedVertices[id]];
+        }
+
+        private void OnDestroy()
+        {
+            vertices_native.Dispose();
+            normals_native.Dispose();
+            remapped_vertices_data.Dispose();
+        }
+
+        public void SetupColliders(SkinnedMeshRenderer keyboardMesh, HPUICustomMesh hpuiCustomMesh)
+        {
+            targetMesh = keyboardMesh;
+            tempMesh = new Mesh(); 
+            targetMesh.BakeMesh(tempMesh, true);
+            
+            if (vertexRemapData == null)
+            {
+                throw new ArgumentException("Missing Vertex Remap Data Asset! Create a new one or provide an existing one!");
+            }
+            
+            if (vertexRemapData.RemappedVertices.Length <= 0)
```

Comment:
why is this a scriptable object? The computation is simple enough it can be computed online, right?

Reply:
The idea is to store different config files for different static meshes. I've changes the implementation to compute and store the remapped data outside play mode, so it doesn't get triggered otherwise

Reply:
If you are storing/caching data, are scriptable objects the right choice? should these not be managed by the package instead of the user?

Still, what is the reason to cache this data?

Reply:
fixed